nexus/src/app/background/init.rs

```diff
@@ -467,29 +476,76 @@ pub mod test {
             SocketAddr::V4(_) => panic!("expected v6 address"),
             SocketAddr::V6(a) => a,
         };
+
+        let update = {
+            use nexus_params::{DnsRecord, Srv};
+
+            let target = "my-great-dns-server.host";
+
+            let mut update = test_dns_update_builder();
+            update.remove_name(internal_dns_srv_name.clone()).unwrap();
+            update
+                .add_name(
+                    internal_dns_srv_name,
+                    vec![
+                        DnsRecord::Srv(Srv {
+                            prio: 0,
+                            weight: 0,
+                            port: new_dns_addr.port(),
+                            target: format!(
+                                "{target}.control-plane.oxide.internal"
+                            ),
+                        }),
+                        DnsRecord::Srv(initial_srv_record.clone()),
+                    ],
+                )
+                .unwrap();
+            update
+                .add_name(
+                    target.to_string(),
+                    vec![DnsRecord::Aaaa(*new_dns_addr.ip())],
+                )
+                .unwrap();
+            update
+        };
+        write_dns_update(&opctx, datastore, update).await;
+        info!(&cptestctx.logctx.log, "updated new dns records");
+
+        // Activate the internal DNS propagation pipeline.
         nexus
-            .upsert_service(
-                &opctx,
-                Uuid::new_v4(),
-                cptestctx.sled_agent.sled_agent.id,
-                Some(Uuid::new_v4()),
-                new_dns_addr,
-                ServiceKind::InternalDns.into(),
-            )
+            .background_tasks
+            .activate(&nexus.background_tasks.task_internal_dns_config);
+
+        wait_propagate_dns(
+            &cptestctx.logctx.log,
+            "initial",
+            initial_dns_dropshot_server.local_addr(),
+            2,
+        )
+        .await;
+
+        let config = dns_config_client
+            .dns_config_get()
             .await
-            .unwrap();
+            .expect("failed to get initial DNS server config");
+        println!("config: {config:#?}");
```

Comment:
Leftover from testing? The `wait_propagate_dns` call right above should validate getting the initial DNS server's config and that it has the right generation at this point.

Reply:
ah, yup, good catch --- I can remove this